modules/openapi-generator/src/main/resources/kotlin-client/libraries/jvm-ktor/infrastructure/ApiClient.kt.mustache

```diff
@@ -199,13 +227,3 @@ import {{packageName}}.auth.*
             RequestMethod.OPTIONS -> HttpMethod.Options
         }
 }
-
-{{#gson}}private class JsonSerializerImpl(private val gson: Gson) : JsonSerializer {
-  override fun write(data: Any, contentType: ContentType): OutgoingContent =
-    ByteArrayContent(gson.toJson(data).toByteArray(StandardCharsets.UTF_8), contentType)
-}{{/gson}}
-
-{{#jackson}}private class JsonSerializerImpl(private val objectMapper: ObjectMapper) : JsonSerializer {
-  override fun write(data: Any, contentType: ContentType): OutgoingContent =
-    ByteArrayContent(objectMapper.writeValueAsBytes(data), contentType)
-}{{/jackson}}
```

Comment:
@Marek00Malik this is not necessary anymore?

Reply:
@4brunu with Ktor 2.x.y introduced a new serialization plugin called ContentNegotiation. The client is responsible for the correct conversion to the type specified by the content type. 
https://ktor.io/docs/serialization-client.html